### PR TITLE
feat(client-debugger): Add container state history in browser debug window

### PR DESCRIPTION
**Pull request by chentong7**
## Description
To be compatible with Chrome Extensions, we need to be able to pass "summaries" (not to be confused with Fluid Summaries) of each Data Object's state across the extension interop boundary. Currently, our prototyped extensibility model is based on rendering visuals; this will need to change.

### CODE REVIEW

**Review by Josmithr — APPROVED** [review]
This looks great! I left 1 comment about how we might simplify the message structure a bit, and there is a documentation tag we need to add to conform to our API-Extractor rules. But otherwise this looks great!

File: packages/tools/client-debugger/client-debugger/api-report/client-debugger.api.md [hunk 1]
```diff
@@ -81,6 +81,18 @@ export interface ContainerStateChangeMessageData extends HasContainerId {
...
 
+// @public
+export interface ContainerStateHistoryMessage extends IDebuggerMessage<ContainerStateHistoryMessageData> {
+    // (undocumented)
+    type: "CONTAINER_STATE_HISTORY";
+}
+
+// @public
+export interface ContainerStateHistoryMessageData extends HasContainerId {
+    // Warning: (ae-incompatible-release-tags) The symbol "history" is marked as @public, but its signature references "ConnectionStateChangeLogEntry" which is marked as @internal
```
[the remark this answers is not in the document]
Comment (chentong7) [reply packages/tools/client-debugger/client-debugger/api-report/client-debugger.api.md:92]: Right right. Modified.

File: packages/tools/client-debugger/client-debugger-chrome-extension/src/react-components/ContainerView.tsx [hunk 2]
```diff
@@ -47,6 +48,9 @@ export function ContainerView(props: ContainerViewProps): React.ReactElement {
 			case PanelView.Telemetry:
 				innerView = <TelemetryView />;
 				break;
+			case PanelView.ContainerStateHistory:
```
[the remark this answers is not in the document]
Comment (chentong7) [reply packages/tools/client-debugger/client-debugger-chrome-extension/src/react-components/ContainerView.tsx:51]: Sure. Ordered

File: packages/tools/client-debugger/client-debugger/src/FluidClientDebugger.ts [hunk 3]
```diff
@@ -231,6 +232,18 @@ export class FluidClientDebugger
 			},
 			this.messageLoggingOptions,
 		);
+
+		postMessageToWindow<ContainerStateHistoryMessage>(
```
[the remark this answers is not in the document]
Comment (chentong7) [reply packages/tools/client-debugger/client-debugger/src/FluidClientDebugger.ts]: Make sense. Let me update `postMessageToWindow` to accept rest parameter.

File: packages/tools/client-debugger/client-debugger/api-report/client-debugger.api.md [hunk 4]
```diff
@@ -50,7 +50,10 @@ export interface ConnectContainerMessage extends IDebuggerMessage<ConnectContain
 // @public
 export type ConnectContainerMessageData = HasContainerId;
 
-// @internal
+// Warning: (ae-incompatible-release-tags) The symbol "ConnectionStateChangeLogEntry" is marked as @public, but its signature references "StateChangeLogEntry" which is marked as @internal
+// Warning: (ae-incompatible-release-tags) The symbol "ConnectionStateChangeLogEntry" is marked as @public, but its signature references "ContainerStateChangeKind" which is marked as @internal
+//
+// @public
```
Comment (alexvy86) [inline packages/tools/client-debugger/client-debugger/api-report/client-debugger.api.md:56]: @chentong7 We'll want to "follow the chain" here and apply the same fix for the new warnings. If we made `ConnectionStateChangeLogEntry` public, we need to make any other types it "depends on" public as well (the types it extends, the types of anything it or one of its methods (if any) might return, etc.). Can you open a PR with those?

**Pull request merged**